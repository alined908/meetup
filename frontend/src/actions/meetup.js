import {GET_MEETUPS_REQUEST, GET_MEETUPS_SUCCESS, GET_MEETUPS_ERROR, 
    ADD_MEETUP, ADD_MEETUP_MEMBER, ADD_EVENT_OPTION, ADD_GLOBAL_MESSAGE, DELETE_MEETUP, EDIT_MEETUP,
    VOTE_MEETUP_EVENT, GET_MEETUP_EVENTS, DELETE_MEETUP_EVENT, ADD_MEETUP_EVENT, EDIT_MEETUP_EVENT} from "../constants/action-types";
import {axiosClient} from '../accounts/axiosClient';
import {history} from '../components/MeetupApp'

export const getMeetups = (data) => async dispatch => {
    dispatch({type: GET_MEETUPS_REQUEST})
    var params;
    if (data.type === "private"){
        params = {type: data.type, ...data.categories && {categories: data.categories}}
    } else {
        params = {
            type: data.type, ...data.categories && {categories: data.categories}, 
            latitude: data.coords.latitude, longitude: data.coords.longitude, 
            radius: data.coords.radius
        }
    }
    
    try {
        const response = await axiosClient.request({
            method: "GET",
            url: "/api/meetups/", 
            headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
            },
            params: params
        })
        setTimeout(() => dispatch({type: GET_MEETUPS_SUCCESS, payload: response.data.meetups}), 100)
    } catch(e){
        console.log(e)
        dispatch({type: GET_MEETUPS_ERROR, payload: e})
    }
}

export const getMeetup = (uri) => async dispatch => {
    try {
        const response = await axiosClient.get(
            `/api/meetups/${uri}/`, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
            }}
        )
        // console.log(response)
        dispatch({type: ADD_MEETUP, payload: response.data})
    } catch(e){
        history.push('/404')
    }
}

export const addMeetup = (formProps, redirectOnSuccess) => async dispatch => {
    var params = {...formProps, 
        date: formProps.date.getFullYear()+ "-" + ("0"+(formProps.date.getMonth()+1)).slice(-2) + "-" + ("0" + formProps.date.getDate()).slice(-2)
    }
    try {
        const response = await axiosClient.post(
            `/api/meetups/`, params, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        dispatch({type: ADD_MEETUP, payload: response.data.meetup})
        console.log(response.data)
        redirectOnSuccess(response.data.meetup.uri)
    } catch(e){
        console.log(e)
    }
} 

export const editMeetup = (formProps, uri, redirectOnSuccess) => async dispatch => {
    console.log('edit meetup')
    var date = new Date(formProps.date)
    var params = {...formProps, 
        date: date.getFullYear()+ "-" + ("0"+(date.getMonth()+1)).slice(-2) + "-" + ("0" + date.getDate()).slice(-2)
    }
    try {
        const response = await axiosClient.patch(
            `/api/meetups/${uri}/`, params, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        dispatch({type: EDIT_MEETUP, payload: response.data})
        console.log(response.data)
        redirectOnSuccess(uri)
    } catch(e){
        console.log(e)
    }
}

export const deleteMeetup = (uri) => async dispatch => {
    try {
        const response = await axiosClient.delete(
            `/api/meetups/${uri}/`, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        console.log(response)
        dispatch({type: DELETE_MEETUP, payload: uri})
        history.push(`/meetups`)
    } catch(e) {
        console.log(e)
    }
}

export const getMeetupEvents = (uri) => async dispatch => {
    try {
        const response = await axiosClient.get(
            `/api/meetups/${uri}/events/`, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        console.log(response)
        dispatch({type: GET_MEETUP_EVENTS, payload: {uri: uri, data: response.data}})
    } catch(e) {
        console.log(e)
    }
}

export const addMeetupEvent = (event) => async dispatch => {
    dispatch({type: ADD_MEETUP_EVENT, payload: event.message})
}

export const deleteMeetupEvent = (event) => async dispatch => {
    dispatch({type: DELETE_MEETUP_EVENT, payload: event.message})
}

export const reloadMeetupEvent = (event) => async dispatch => {
    dispatch({type: EDIT_MEETUP_EVENT, payload: event.message})
}

export const voteMeetupEvent = (event) => async dispatch => {
    dispatch({type: VOTE_MEETUP_EVENT, payload: event.message})
}
    
export const decideMeetupEvent = (event) => async dispatch => {
    dispatch({type: EDIT_MEETUP_EVENT, payload: event.message})
}

export const addMeetupMember = (event) => async dispatch => {
    dispatch({type: ADD_MEETUP_MEMBER, payload: event.message})
}

export const addEventOption = (event) => async dispatch => {
    dispatch({type: ADD_EVENT_OPTION, payload: event.message})
}

export const sendMeetupEmails = (uri) => async dispatch => {

    try {
        await axiosClient.post(
            `/api/meetups/${uri}/email/`, {}, {headers: {
                "Authorization": `Bearer ${localStorage.getItem('token')}`
        }})
        dispatch({type: ADD_GLOBAL_MESSAGE, payload: {type: "success", message: "Successfully sent emails"}})
    } catch(e) {
        dispatch({type: ADD_GLOBAL_MESSAGE, payload: {type: "error", message: "Unable to send emails"}})
    }
}