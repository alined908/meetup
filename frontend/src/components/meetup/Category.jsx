import React from 'react'
import {Button} from "@material-ui/core"

const Category = (props) => {
    const [variant, setVariant] = React.useState(props.clicked)

    const handleClick = () => {
        setVariant(variant === 'outlined' ? 'contained' : 'outlined')
    }

    return (
        <Button variant={variant} color="primary" onClick={() => handleClick()}>{props.category.label}</Button>
    )
}

export default Category