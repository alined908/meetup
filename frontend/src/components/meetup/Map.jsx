import React, {Component} from "react"
import ReactMapGL, {Marker, NavigationControl, FlyToInterpolator} from "react-map-gl"
import LocationOnIcon from '@material-ui/icons/LocationOn';
import * as d3 from 'd3-ease'

const token = "pk.eyJ1IjoiZGFuaWVsOTA4OTA4IiwiYSI6ImNrNnR6aTdtbTA0bmwzbW1ya3FncjdlcGkifQ.cBAvCpS5zDNwzyb_GUZaoA"

class Map extends Component {

    constructor(props) {
        super(props);
        this.state = {
          viewport: {
            latitude: this.props.location[0],
            longitude: this.props.location[1],
            zoom: 12.5,
            bearing: 0,
            pitch: 0
          }
        };
    }

    onLoad = () => {
        const viewport = {
            ...this.state.viewport,
            zoom: 15,
            transitionDuration: 6000,
            transitionInterpolator: new FlyToInterpolator(),
            transitionEasing: d3.easeSin
        }
        this.setState({viewport})
    }


    render() {
        return (
          <ReactMapGL
            {...this.state.viewport}
            width="450px"
            height="350px"
            mapStyle="mapbox://styles/mapbox/light-v9"
            onViewportChange={viewport => this.setState({viewport})}
            onLoad={this.onLoad}
            mapboxApiAccessToken={token}
          >
            <div style={{position: 'absolute', right: 25, top: 10}}>
                <NavigationControl showCompass={false}/>
            </div>
            <Marker latitude={this.props.location[0]} longitude={this.props.location[1]}>
                <LocationOnIcon fontSize="large" color="secondary"></LocationOnIcon>
            </Marker>
          </ReactMapGL>
        );
      }
}

export default Map