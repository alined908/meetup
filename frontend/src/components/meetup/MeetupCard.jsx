import React from 'react'
import {makeStyles, Typography, Paper, Badge, Tooltip, Avatar} from '@material-ui/core'
import {Today as TodayIcon, Notifications as NotificationsIcon, Room as RoomIcon} from '@material-ui/icons'
import {GroupAvatars} from "../components"
import {Link} from 'react-router-dom'
import moment from 'moment';
import Skeleton from '@material-ui/lab/Skeleton';
import {meetupPropType} from "../../constants/prop-types"
import styles from '../../styles/meetup.module.css'

const useStyles = makeStyles({
    paper: {
        width: "100%",
        height: "100%",
        minWidth: "320px",
    },
    middle: {
        height: "150px",
        margin: "1rem 0",
    }
})

const MeetupCard = ({meetup}) => {
    const classes = useStyles();
    const members = Object.values(meetup.members)
    const users = []
    members.map((member) => users.push(member.user))

    return (
        <>
        {meetup ? 
            <Link to={`/meetups/${meetup.uri}`}>
                <Paper elevation={6} square className={classes.paper}>
                    <div className={styles.cardInner}>
                        <div className={styles.cardTop}> 
                            <div className={`${styles.cardIcon} ${styles.flexClip2}`}>
                                <Tooltip title={meetup.name}>
                                    <Typography className={styles.flexEllipse} variant="h5">
                                        {meetup.name}
                                    </Typography>
                                </Tooltip>
                            </div>
                            <div className={styles.cardIcon}>
                                <TodayIcon/>
                                {moment(meetup.date).local().format("dddd, MMMM D")}
                            </div>
                            {meetup.notifs !== null && meetup.notifs > 0 && <Badge badgeContent={meetup.notifs} color="primary">
                                <NotificationsIcon />
                            </Badge>
                            }  
                        </div>
                        {meetup.categories.length === 0 ?
                            <div className={classes.middle}>
                                <Skeleton animation="wave" variant="rect" height="100%"/>
                            </div> :
                            <div className={`${classes.middle} ${styles.cardEvents}`} >
                                <div className={styles.cardEventsHeader}>Events</div>
                                <div className={styles.cardEventsCategories}>
                                    {meetup.categories.slice(0, 4).map((category) => 
                                        <div key={category.id} className={styles.cardEventsCategory}>
                                            <Avatar style={{width: 28, height: 28}} variant="square"
                                                src={`${process.env.REACT_APP_S3_STATIC_URL}${category.api_label}.png`} 
                                            />
                                            {category.label}
                                        </div>
                                    )}
                                </div>
                            </div>
                        }
                        <div className={styles.cardBottom}>
                            <Tooltip title={meetup.location}>
                                <div className={`${styles.cardIcon} ${styles.flexClip}`}>
                                    <RoomIcon/>
                                    <div className={styles.flexEllipse}>
                                        {meetup.location}
                                    </div>
                                </div>
                            </Tooltip>
                            <div className={`${styles.cardIcon} ${styles.flexClip}`}>
                                <GroupAvatars members={users}/>
                                <span className={`${styles.flexEllipse} ${styles.cardMembers}`}>
                                    {users.length + " member"}{users.length > 1 ? "s" : ""}
                                </span>
                            </div>
                        </div>
                    </div>
                </Paper>
            </Link> : 
            <Paper className={classes.paper} elevation={3}>
                <Skeleton variant="rect" animation="wave"/>
            </Paper> 
        }
        </>
    )
}

MeetupCard.propTypes = {
    meetup: meetupPropType
}

export default MeetupCard