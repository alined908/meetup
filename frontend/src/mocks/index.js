import moment from 'moment'

//Category
export const category = {
    id: 1, 
    image: null,
    label: "Dessert", 
    api_label: "desserts"
}

//User
export const user = {
    id: 1, 
    email: "daniel@gmail.com", 
    first_name: "Daniel", 
    last_name: "Lee",
    avatar: null
}
export const user2 = {
    id: 2, 
    email: "example2@gmail.com", 
    first_name: "Bob", 
    last_name: "Jim", 
    avatar: null
}
export const user3 = {
    id: 3, 
    email: "example3@gmail.com", 
    first_name: "Bob", 
    last_name: "Jim", 
    avatar: null
}
export const settings = {
    radius: 25, 
    location: null, 
    latitude: null,
    longitude: null
}

export const userWithSettings = {
    ...user,
    settings: {...settings}
}
export const friend = {
    id: 1, 
    user: user2, 
    created_at: moment().format(), 
    chat_room: "something"
}
export const friends = [friend]
export const preference = {
    id: 1, 
    user, 
    category, 
    name: "Dessert", 
    ranking: 1, 
    timestamp: moment().format()
}
export const preferences = [preference]
export const friendInvite = {
    id: 1, 
    timestamp: moment().format(), 
    status: 1, 
    uri: "random", 
    sender: user, 
    receiver: user2
}

//Chat
export const message = {
    id: 1, 
    message: "hello", 
    timestamp: moment().format(), 
    room_id: 1, 
    sender: user, 
    is_notif: false
}
export const message2 = {
    id: 2, 
    message: "hello", 
    timestamp: moment().format(), 
    room_id: 1, 
    sender: user, 
    is_notif: true
}
export const messages = [message, message2]
export const members = {1: user, 2: user2}
export const room = {
    uri: {
        id: 1, 
        uri: "abc", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: 1, 
        meetup:null, 
        notifs: 0
    }
}
export const rooms = {
    friend: {
        id: 1, 
        uri: "abc", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: 1, 
        meetup:null, 
        notifs: 0
    },
    meetup: {
        id: 2, 
        uri: "xyz", 
        name: "Room", 
        timestamp: moment().format(), 
        members: members, 
        friendship: null, 
        meetup:1, 
        notifs: 3
    }
}

//General
export const error = {
    message: "Something went wrong."
}

//Meetup
export const meetupMember1 = {
    1: {
        id: 1,
        user,
        ban: false,
        admin: true
    }
}

export const meetupMember2 = {
    2: {
        id: 2,
        user: user2,
        ban: false,
        admin: false
    }
}

export const meetupMember3 = {
    3: {
        id: 3,
        user: user3,
        ban: false,
        admin: false
    }
}

export const meetupMembers = {...meetupMember1, ...meetupMember2}

export const meetup = {
    uri: {
        id: 1, 
        name: "Meetup", 
        uri: "uri", 
        creator: user, 
        location: "location", 
        date: moment().format("YYYY-MM-DD"),
        members: meetupMembers,
        notifs: 0,
        public: true,
        categories: [category],
        latitude: 34.228754, 
        longitude: -118.2351192,
        events: {},
        isMeetupEventsInitialized : true
    }
}

export const meetups = {...meetup}

export const event = {
    1: {
        id: 1,
        meetup: 1,
        creator: user,
        title: "Dessert",
        start: moment().format(),
        end: moment().format(),
        chosen: null,
        categories: [category],
        options: {},
        price: "1, 2",
        distance: 16000,
        entries: {},
        random: true
    }
}
export const events = {...event}

export const option = {
    1 : {
        id: 1,
        score: 0,
        option: null,
        restaurant: {},
        votes: {},
        banned: false
    }
}

export const options = {...option}

export const tokens = {
    access: "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoiYWNjZXNzIiwiZXhwIjoxNTg4MjI5OTEzLCJqdGkiOiIwOTg2YTg0MDA1ZGM0OGQ1ODEzYmIyNzE1OWViNGUyZCIsInVzZXJfaWQiOjEsInVzZXIiOnsiaWQiOjEsImVtYWlsIjoiZGFuaWVsQGdtYWlsLmNvbSIsImZpcnN0X25hbWUiOiJEYW5pZWwiLCJsYXN0X25hbWUiOiJMZWUiLCJhdmF0YXIiOm51bGwsInNldHRpbmdzIjp7InJhZGl1cyI6MjUsImxvY2F0aW9uIjpudWxsLCJsYXRpdHVkZSI6bnVsbCwibG9uZ2l0dWRlIjpudWxsfX19.FbffRthOKX5F0TGvqA5mZXuXUWNkUojzKexDGCoP518",
    refresh : "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ0b2tlbl90eXBlIjoicmVmcmVzaCIsImV4cCI6MTU4ODM4Mjk0NiwianRpIjoiN2ZhNGY2MDE4OWM3NDA5ZmFjOTMwYmE4ZTRlMWFhMTMiLCJ1c2VyX2lkIjoxLCJ1c2VyIjp7ImlkIjoxLCJlbWFpbCI6ImRhbmllbEBnbWFpbC5jb20iLCJmaXJzdF9uYW1lIjoiRGFuaWVsIiwibGFzdF9uYW1lIjoiTGVlIiwiYXZhdGFyIjpudWxsLCJzZXR0aW5ncyI6eyJyYWRpdXMiOjI1LCJsb2NhdGlvbiI6bnVsbCwibGF0aXR1ZGUiOm51bGwsImxvbmdpdHVkZSI6bnVsbH19fQ.qqaY4Ye7rmyOKLbDS2EKz4SGWH0fwZ0MvrGP6NJ3IRE"
}