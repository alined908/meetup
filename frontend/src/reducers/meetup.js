import {GET_MEETUPS, ADD_MEETUP, DELETE_MEETUP, CLEAR_STORE, VOTE_MEETUP_EVENT, ADD_MEETUP_EVENT, GET_MEETUP_EVENTS, DELETE_MEETUP_EVENT, EDIT_MEETUP_EVENT} from '../constants/action-types';

const defaultState = {
    meetups: {},
    activeMeetup: null,
    isMeetupsInitialized: false,
    isMeetupEventsInitialized: false,
}

export default function meetupReducer(state = defaultState, action){
    switch (action.type){
        case GET_MEETUPS:
            return {...state, meetups: action.payload, isMeetupsInitialized: true}
        case ADD_MEETUP:
            return {...state, meetups: {...state.meetups, [action.payload.uri]: action.payload}}
        case DELETE_MEETUP:
            var meetups = {}
            Object.keys(state.meetups).map((key) => {
                if (key !== action.payload){
                    meetups[key] = state.meetups[key]
                }
            })
            return {...state, meetups: meetups}
        case GET_MEETUP_EVENTS:
           var meetups = {}
           const data = action.payload.data
           Object.keys(state.meetups).map((key) => 
               (key === action.payload.uri) ? meetups[key] = ({...state.meetups[key], events: data}) : (meetups[key] = state.meetups[key])
           )
            return {...state, meetups: meetups, isMeetupEventsInitialized: true}
        case ADD_MEETUP_EVENT:
            return {...state, 
                        meetups: {...state.meetups, 
                            [action.payload.meetup]: {...state.meetups[action.payload.meetup], 
                                events: {...state.meetups[action.payload.meetup].events, 
                                    ...action.payload.event}}}}
        case EDIT_MEETUP_EVENT:
            const event_id = action.payload.event_id
            return {...state, 
                        meetups: {...state.meetups, 
                            [action.payload.meetup]: {...state.meetups[action.payload.meetup],
                                events: {...state.meetups[action.payload.meetup].events, [event_id]: action.payload.event}}}}
        case VOTE_MEETUP_EVENT:
            var meetup = action.payload.meetup
            return {...state, meetups: {...state.meetups, 
                            [meetup]: {...state.meetups[meetup], 
                                events: {...state.meetups[meetup].events, 
                                    [action.payload.event]: {...state.meetups[meetup].events[action.payload.event], 
                                        options: {...state.meetups[meetup].events[action.payload.event].options, 
                                            [action.payload.option_id]: action.payload.option
                    }}}}}}
        case DELETE_MEETUP_EVENT:
            var events = {}
            const uri = action.payload.uri
            Object.keys(state.meetups[uri].events).map((key) => {
                if (key !== action.payload.event.toString()){
                    events[key] = state.meetups[uri].events[key]
                }
            })
            return {...state, meetups: {...state.meetups, [uri] : {...state.meetups[uri], events: events}}}
        case CLEAR_STORE:
            return defaultState
        default:
            return state
    }
}