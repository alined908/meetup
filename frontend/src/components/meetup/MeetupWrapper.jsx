import React, {Component} from "react"
import {connect} from "react-redux";
import {getMeetup} from "../../actions"
import {Meetup} from '../components';
import {Redirect} from 'react-router-dom'
import {CircularProgress} from '@material-ui/core'

class MeetupWrapper extends Component {
    constructor(props){
        super(props)
        const uri = this.props.match.params.uri
        if (!(uri in this.props.meetups)){
            this.props.getMeetup(uri)
        }
    }

    determineIsUserMember = (members) => {
        let isUserMember = false;
        for (let key of Object.keys(members)){
            const member = members[key]
            if (member.user.id === this.props.user.id){
                return true
            }
        }
        return isUserMember
    }

    render () {
        const meetup = this.props.meetups[this.props.match.params.uri]
        const isUserMember = meetup && this.determineIsUserMember(meetup.members)

        return (    
            <>
                {meetup ?
                    ((isUserMember || meetup.public) ? 
                        <Meetup key={meetup.id} meetup={meetup} isUserMember={isUserMember}/> : 
                        <Redirect to="/meetups"/>
                    ) :
                    <div className="loading">
                        <CircularProgress/>
                    </div>
                }
            </>
        )
    }
}

function mapStateToProps(state) {
    return {
        meetups: state.meetup.meetups,
        user: state.user.user
    }
}

const mapDispatchToProps = {
    getMeetup
}

export default connect(mapStateToProps, mapDispatchToProps)(MeetupWrapper)