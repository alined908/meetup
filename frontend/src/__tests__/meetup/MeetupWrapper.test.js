import React from 'react'
import {shallow} from 'enzyme';
import {UnderlyingMeetupWrapper} from "../../components/meetup/MeetupWrapper"
import * as mocks from "../../mocks"

const props = {
    meetups: mocks.meetups,
    user: mocks.user,
    getMeetup: jest.fn(),
    match: {params: {uri: "uri"}}
}

describe("MeetupWrapper tests", () => {

    afterEach(() => {
        jest.clearAllMocks();
    });

    it ('should render without crashing', () => {
        const wrapper = shallow(<UnderlyingMeetupWrapper {...props}/>)
        expect(wrapper).toMatchSnapshot()
    })

    it ('should render loading and call getMeetup if uri not in meetups', () => {
        const newProps = {...props, match: {params: {uri: "uri3"}}}
        const wrapper = shallow(<UnderlyingMeetupWrapper {...newProps}/>)
        expect(props.getMeetup.mock.calls.length).toBe(1)
        expect(wrapper.find('.loading')).toHaveLength(1)
    })

    it ('should redirect if user is not member and meetup is not public' , () => {
        const newProps = {...props, user: mocks.user3, meetups: mocks.privateMeetup, match: {params: {uri: "uri2"}}}
        const wrapper = shallow(<UnderlyingMeetupWrapper {...newProps}/>)
        expect(wrapper.find("Redirect")).toHaveLength(1)
    })
})