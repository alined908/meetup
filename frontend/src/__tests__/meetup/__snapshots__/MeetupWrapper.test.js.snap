// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MeetupWrapper tests should render without crashing 1`] = `
<Fragment>
  <Connect(Meetup)
    isUserMember={true}
    key="1"
    meetup={
      Object {
        "categories": Array [
          Object {
            "api_label": "desserts",
            "id": 1,
            "image": null,
            "label": "Dessert",
          },
        ],
        "creator": Object {
          "avatar": null,
          "email": "daniel@gmail.com",
          "first_name": "Daniel",
          "id": 1,
          "last_name": "Lee",
        },
        "date": "2020-05-10",
        "events": Object {},
        "id": 1,
        "isMeetupEventsInitialized": true,
        "latitude": 34.228754,
        "location": "location",
        "longitude": -118.2351192,
        "members": Object {
          "1": Object {
            "admin": true,
            "ban": false,
            "id": 1,
            "user": Object {
              "avatar": null,
              "email": "daniel@gmail.com",
              "first_name": "Daniel",
              "id": 1,
              "last_name": "Lee",
            },
          },
          "2": Object {
            "admin": false,
            "ban": false,
            "id": 2,
            "user": Object {
              "avatar": null,
              "email": "example2@gmail.com",
              "first_name": "Bob",
              "id": 2,
              "last_name": "Jim",
            },
          },
        },
        "name": "Meetup",
        "notifs": 0,
        "public": true,
        "uri": "uri",
      }
    }
  />
</Fragment>
`;
